code/parse/parselo.cpp

```diff
@@ -2258,6 +2260,74 @@ void read_raw_file_text(const char *filename, int mode, char *raw_text)
 	cfclose(mf);
 }
 
+// Goober5000, based partly on above iconv usage
+void coerce_to_utf8(SCP_string &buffer, const char *str)
+{
+	auto len = strlen(str);
+
+	// Validate the UTF-8 encoding
+	auto invalid = utf8::find_invalid(str, str + len);
+	if (invalid == str + len)
+	{
+		// turns out this is valid UTF-8
+		buffer.assign(str);
+		return;
+	}
+
+	bool isLatin1 = util::guessLatin1Encoding(str, len);
+
+	// we can convert it
+	if (isLatin1)
+	{
+		size_t newlen = len;
+		auto newstr = new char[newlen];
```

Comment:
Please use `std::unique_ptr<char[]>` for a bit more safety.

Reply:
Ok.